test/server-unit/mysql_functions.spec.js

```diff
@@ -0,0 +1,133 @@
+/* eslint global-require:off */
+const { expect } = require('chai');
+
+function DatabaseUnitTests() {
+  let db;
+  before(() => {
+    db = require('../../server/lib/db');
+  });
+
+  const tableName1 = 'my_test_table1';
+  const tableName2 = 'my_test_table2';
+  const constraint1 = 'constraint1';
+  const constraint2 = 'constraint2';
+
+  const createTable1Sql = `
+    CREATE TABLE IF NOT EXISTS ${tableName1} (
+      uuid BINARY(16) PRIMARY KEY,
+      name VARCHAR(40)
+    )ENGINE=InnoDB DEFAULT CHARSET=latin1;
+  `;
+
+  const createTable2Sql = `
+    CREATE TABLE IF NOT EXISTS ${tableName2} (
+      uuid BINARY(16) PRIMARY KEY,
+      type VARCHAR(40),
+      uuid_parent BINARY(16),
+      uuid_parent2 BINARY(16),
+      INDEX \`type\`(\`type\`),
+    CONSTRAINT \`${constraint1}\` FOREIGN KEY (\`uuid_parent\`)  
+    REFERENCES \`${tableName2}\` (\`uuid\`) ON DELETE CASCADE,
+
+    CONSTRAINT \`${constraint2}\` FOREIGN KEY (\`uuid_parent2\`)
+    REFERENCES \`${tableName2}\` (\`uuid\`) ON DELETE CASCADE
+    )ENGINE=InnoDB DEFAULT CHARSET=latin1;
+  `;
+
+  const checkIfConstraintExistSql = ` SELECT Constraint_exists("${tableName2}", "${constraint1}") as exist;`;
+
+  const checkIfConstraintIfExistSql = ` SELECT Constraint_exists("${tableName2}", "unknown") as exist;`;
+  const dropFKExistSql = ` SELECT Constraint_exists("${tableName2}",  "${constraint2}") as exist;`;
+  const dropForeignKeySql = ` CALL drop_foreign_key("${tableName2}", "${constraint2}");`;
+  const dropUnknownForeignKeySql = ` CALL drop_foreign_key("${tableName2}", "unknown");`;
+  const dropUnknownColumnSql = ` CALL drop_column_if_exists("${tableName2}", "unknown");`;
+  const dropColumnSql = ` CALL drop_column_if_exists("${tableName2}", "uuid_parent2");`;
+  const checkIfColumnExistSql = ` SELECT column_exists("${tableName2}", "uuid_parent2") as exist;`;
+
+  const checkIfIndexExistSql = ` SELECT index_exists("${tableName2}", "type") as exist;`;
+  const dropUnknownIndexSql = ` CALL drop_index_if_exists("${tableName2}", "unknown");`;
+  const dropIndexSql = ` CALL drop_index_if_exists("${tableName2}", "type");`;
+
+  it('#should create the test table1', async () => {
+    await db.exec(createTable1Sql);
+    expect(true).to.equal(true);
```

Comment:
I don't understand how this test actually checks to see if the table was created. 
Couldn't you check the return value of 'await db.exec()' which should produce the error flag of the MySQL query call, correct?   So would this work?

  const result = await db.exec(createTable1Sql);
  expect(result).to.equal(true);

or something similar?

Reply:
The additions look good to me!